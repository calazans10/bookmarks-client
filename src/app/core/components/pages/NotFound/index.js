import React from 'react';
import PropTypes from 'prop-types';
import MainWrapper from '../../modules/MainWrapper';
import MainContent from '../../modules/MainContent';
import PageSection from '../../modules/PageSection';
import ButtonLink from '../../modules/ButtonLink';

const NotFound = ({ history }) => (
  <MainWrapper>
    <MainContent title="Page not found">
      <PageSection title="404 - Page not found">
        <p>Sorry, the page you were looking for cannot be found!</p>
        <p>
          Do you want to go back to where you were?{' '}
          <ButtonLink onClick={() => history.goBack()}>Click here</ButtonLink>.
        </p>
      </PageSection>
    </MainContent>
  </MainWrapper>
);

NotFound.propTypes = {
  history: PropTypes.object.isRequired,
};

export default NotFound;
