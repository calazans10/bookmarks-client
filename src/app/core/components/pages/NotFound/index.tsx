import React from 'react';
import { useHistory } from 'react-router-dom';
import MainWrapper from '../../modules/MainWrapper';
import MainContent from '../../modules/MainContent';
import PageSection from '../../modules/PageSection';
import ButtonLink from '../../modules/ButtonLink';

const NotFound = () => {
  const history = useHistory();

  return (
    <MainWrapper>
      <MainContent>
        <PageSection title="404 - Page not found">
          <p>Sorry, the page you were looking for cannot be found!</p>
          <p>
            Do you want to go back to where you were?{' '}
            <ButtonLink onClick={() => history.goBack()}>Click here</ButtonLink>.
          </p>
        </PageSection>
      </MainContent>
    </MainWrapper>
  );
};

export default NotFound;
