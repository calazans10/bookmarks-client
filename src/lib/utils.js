export const isPaginationVisible = (count, limit, total) => count <= total && total > limit;
