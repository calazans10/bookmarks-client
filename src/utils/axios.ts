import axios from 'axios';

const enhanceError = error => {
  const defaultMessage = 'The operation could not be completed. Please try again later.';
  const enhancedError = error || {};

  if (!enhancedError.response) {
    enhancedError.response = {};
    enhancedError.response.message = defaultMessage;
  }

  const { status } = enhancedError.response;
  if (!status) {
    enhancedError.response.status = 0;
  }

  return enhancedError;
};

class AxiosFactory {
  createInstance(url) {
    const source = axios.CancelToken.source();
    this.instance = axios.create({
      baseURL: url,
      timeout: process.env.REACT_APP_TIMEOUT,
      cancelToken: source.token,
    });

    this.instance.interceptors.request.use(
      config => {
        const token = sessionStorage.getItem('token');
        if (token) {
          config.headers.Authorization = `Bearer ${token}`;
        }

        setTimeout(() => {
          source.cancel();
        }, process.env.REACT_APP_TIMEOUT + 1000);

        return config;
      },
      error => {
        const enhancedError = enhanceError(error);
        Promise.reject(enhancedError);
      }
    );

    this.instance.interceptors.response.use(
      config => config,
      error => {
        const enhancedError = enhanceError(error);
        return Promise.reject(enhancedError);
      }
    );

    return this.instance;
  }
}

export default new AxiosFactory();
