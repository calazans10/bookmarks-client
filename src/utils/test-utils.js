import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import rootReducer from '../reducers';

export function renderWithRedux(
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
