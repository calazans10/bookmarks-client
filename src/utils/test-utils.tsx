import React, { ReactNode } from 'react';
import { AnyAction, Action, createStore, Store } from 'redux';
import { Provider } from 'react-redux';
import { render, RenderResult } from '@testing-library/react';
import rootReducer from '../reducers';

interface RenderWithRedux<
  S = any,
  A extends Action = AnyAction,
  I extends S = any
  > {
  (
    ui: ReactNode,
    reduxOptions?: {
      store?: Store<S, A>
      initialState?: I
    }
  ): RenderResult & {
    store: Store<S, A>
  }
}

export const renderWithRedux: RenderWithRedux = (
  ui,
  { initialState, store = createStore(rootReducer, initialState) } = {}
) => {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}
