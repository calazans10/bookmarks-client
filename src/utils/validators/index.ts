// eslint-disable-next-line no-useless-escape
const EMAIL_REGEX = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
const PASSWORD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
const NUMERIC_REGEX = /^[0-9]+$/;
const FULLNAME_REGEX = /^(.+?) ([^\s,]+)(,? (?:[JS]r\.?|III?|IV))?$/i;
const URL_REGEX = /(^|\s)((https?:\/\/)?[\w-]+(\.[\w-]+)+\.?(:\d+)?(\/\S*)?)/gi;

const required = value => (value ? undefined : 'É necessário preencher este campo.');

const mustChooseOne = value =>
  typeof value === 'undefined' || value.length === 0
    ? 'É necessário escolher ao menos uma opção.'
    : undefined;

const mustBeEmail = value => {
  if (!value || EMAIL_REGEX.test(value)) {
    return undefined;
  }

  return 'O e-mail informado não é válido.';
};

const mustBePassword = value => {
  if (!value || PASSWORD_REGEX.test(value)) {
    return undefined;
  }

  return 'Senha inválida. Use oito ou mais caracteres contendo no mínimo, uma letra maiúscula, uma letra minúscula e um número.';
};

const mustBeNumeric = value => {
  if (!value || NUMERIC_REGEX.test(value.replace(/\W/g, ''))) {
    return undefined;
  }

  return 'Este campo só aceita números.';
};

const mustBeUrl = value => {
  if (!value || URL_REGEX.test(value)) {
    return undefined;
  }

  return 'Digite um endereço válido.';
};

const mustBeFullName = value => {
  if (!value || FULLNAME_REGEX.test(value)) {
    return undefined;
  }

  return 'Digite o nome completo.';
};

const minLength = min => value => {
  if (!value || value.replace(/\W/g, '').length >= min) {
    return undefined;
  }

  return `Este campo requer no mínimo ${min} ${min === 1 ? 'caractere' : 'caracteres'}.`;
};

const maxLength = max => value => {
  if (!value || value.replace(/\W/g, '').length <= max) {
    return undefined;
  }

  return `Este campo requer no máximo ${max} ${max === 1 ? 'caractere' : 'caracteres'}.`;
};

const composeValidators = (...validators) => value =>
  validators.reduce((error, validator) => error || validator(value), undefined);

export {
  required,
  mustChooseOne,
  mustBeEmail,
  mustBePassword,
  mustBeNumeric,
  mustBeUrl,
  mustBeFullName,
  minLength,
  maxLength,
  composeValidators,
};
