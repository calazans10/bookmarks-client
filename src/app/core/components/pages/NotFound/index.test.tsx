import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { renderWithRedux } from 'utils/test-utils';
import NotFound from '.';

describe('NotFound', () => {
  it('renders without crashing', () => {
    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/unknown']}>
        <NotFound />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
