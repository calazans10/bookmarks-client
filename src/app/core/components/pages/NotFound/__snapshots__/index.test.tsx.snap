// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders without crashing 1`] = `
<div
  class="sc-AxhUy jsxOVu"
>
  <div
    class="sc-AxjAm gVcqXk"
  >
    <p
      class="sc-AxirZ dQMFuN"
    >
      <span
        class="sc-AxiKw cGiMxH"
      />
       
      
    </p>
    <button
      aria-label="Close alert"
      class="sc-AxhCb fbaVap"
      type="button"
    >
      <span>
        Close
      </span>
    </button>
  </div>
  <main
    class="sc-AxgMl fYQEfX"
  >
    <section
      class="sc-AxheI kcxa-dV"
    >
      <h1
        class="sc-Axmtr leJLpf"
      >
        404 - Page not found
      </h1>
      <p>
        Sorry, the page you were looking for cannot be found!
      </p>
      <p>
        Do you want to go back to where you were?
         
        <button
          class="sc-AxmLO kWGaqt"
          type="button"
        >
          Click here
        </button>
        .
      </p>
    </section>
  </main>
</div>
`;
